Automattic/wp-calypso
# Partner Portal: Refactor key selection to use the http() utilities and to look better

Context:
* Project thread: pbtFPC-Um-p2
* i4 designs: pbtFPC-103-p2

#### Changes proposed in this Pull Request

* In the Partner Portal, improve partner key selection UI visually.
* In the Partner Portal, refactor key selection data layer to use the http() utilities.

#### Testing instructions

* If you do not have a partner key, please contact Infinity for one or you will be unable to view the UI.
* Checkout PR locally, run `yarn && yarn start-jetpack-cloud`.
* Visit http://jetpack.cloud.localhost:3000/partner-portal and select your partner key, if prompted.
* Make sure you are shown the error message (shown at the bottom) if you have no active partner keys.
* Make sure key selection shows up if you have multiple active partner keys.
* Make sure key selection is skipped if you have a single active partner key (it should get autoselected).

Key selection:
![Screenshot 2021-03-03 at 21 15 22](https://user-images.githubusercontent.com/22746396/109859384-93e91800-7c65-11eb-9a97-96e87fd22ba8.png)

Missing/error-ed out key selection:
![Screenshot 2021-03-03 at 21 17 47](https://user-images.githubusercontent.com/22746396/109859665-e9252980-7c65-11eb-970e-54d410101a32.png)
- This screen will show up whenever keys cannot be loaded (e.g. user is not a partner, does not have keys or request failed).
- The error toast message will only appear when the request failed.
